cmd/run_internal_test.go

```diff
@@ -66,3 +66,97 @@ func TestBindPrivKeyFlag(t *testing.T) {
 		})
 	}
 }
+
+func TestBindRunFlagsValidation(t *testing.T) {
+	tempDir := t.TempDir()
+	certFile, err := os.CreateTemp(tempDir, "cert.pem")
+	require.NoError(t, err)
+	keyFile, err := os.CreateTemp(tempDir, "cert.key")
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		require.NoError(t, os.Remove(certFile.Name()))
+		require.NoError(t, os.Remove(keyFile.Name()))
+	})
+
+	tests := []struct {
+		Name      string
+		Args      []string
+		AppConfig *app.Config
+		Err       string
+	}{
+		{
+			Name: "minimum flags set",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node"),
+		},
+		{
+			Name: "too long nickname",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--nickname", "thisnicknameiswaytoolongandshouldfail"),
+			Err:  "flag 'nickname' can not exceed 32 characters",
+		},
+		{
+			Name: "valid nickname",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--nickname", "validnickname"),
+		},
+		{
+			Name: "too long graffiti bytes",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--graffiti", "thisgraffitostringiswaytoolongandshouldfail"),
+			Err:  "graffiti string length is greater than maximum size",
+		},
+		{
+			Name: "valid graffiti bytes length",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--graffiti", "validgraffiti"),
+		},
+		{
+			Name: "invalid beacon node headers separator",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--beacon-node-headers", "key1=value1,key2:value2"),
+			Err:  "beacon node headers must be comma separated values formatted as header=value",
+		},
+		{
+			Name: "invalid beacon node headers completeness",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--beacon-node-headers", "key1=value1,key2="),
+			Err:  "beacon node headers must be comma separated values formatted as header=value",
+		},
+		{
+			Name: "valid beacon node headers",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--beacon-node-headers", "key1=value1,key2=value2"),
+		},
+		{
+			Name: "vc tls cert set without key",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--vc-tls-cert-file", "cert.pem"),
+			Err:  "both VCTLSCertFile and VCTLSKeyFile must be set or both must be empty",
+		},
+		{
+			Name: "vc tls key set without cert",
+			Args: slice("run", "--beacon-node-endpoints", "http://beacon.node", "--vc-tls-key-file", "cert.key"),
```

Comment:
My flag and description changes break a couple of these tests, not making suggestions for all as I'm sure you can figure them out :)